web/app.js

```diff
@@ -675,6 +675,9 @@ var PDFViewerApplication = {
       return;
     }
 
+    // There could be a filename available from network call
+    filename = this.pdfDocumentProperties.getFileName();
```

Comment:
I am not sure at that point an updated file name from the network call is available. User could click on download link while the network call is in progress, that is the reason why we download directly from the URL in some cases here, right?